### PR TITLE
feat: Documentation reference check

### PR DESCRIPTION
**Pull request by mateusoliveira43**
Add documentation reference check to CI. This allows `.golangci.reference.yml` file to be always up to date with code.

Related to https://github.com/golangci/golangci-lint/issues/2645

### CODE REVIEW

File: .golangci.reference.yml [hunk 1]
```diff
@@ -1505,6 +1523,14 @@ linters-settings:
     # Default: 0.8
     confidence: 0.1
 
+    error-code: 1
```
Comment (mateusoliveira43) [inline .golangci.reference.yml:1526]: some of the options I was not able to get help/documentation from where I found them in the code

File: pkg/config/reader.go [hunk 2]
```diff
@@ -99,7 +165,21 @@ func (r *FileReader) parseConfig() error {
 
 		// Needed for forbidigo.
 		mapstructure.TextUnmarshallerHookFunc(),
-	))); err != nil {
+	)), func(config *mapstructure.DecoderConfig) {
+		config.ErrorUnused = true
+		if os.Getenv("HELP_RUN") == "2" {
```
Comment (mateusoliveira43) [inline pkg/config/reader.go:170]: I had cyclic imports errors here. Maybe add all envVars names to a single file? related to https://github.com/golangci/golangci-lint/pull/3196

**Comment by ldez** [comment]
Please open an issue before doing something like that.

**Comment by mateusoliveira43** [comment]
@ldez created the issue. Please contact me if you have any questions, I think this can increase the product quality

**Pull request closed without merging**